gateway/webhook/webhookTransformer.go

```diff
@@ -45,11 +50,53 @@ func (bt *batchWebhookTransformerT) markResponseFail(reason string) transformerR
 	return resp
 }
 
+func (bt *batchWebhookTransformerT) transformUsingPlugin(plugin plugins.Plugin, payload []byte, sourceType string) transformerBatchResponseT {
+	transformStart := time.Now()
+	var events []map[string]interface{}
+	err := json.Unmarshal(payload, &events)
+	if err != nil {
+		bt.stats.failedStat.Count(len(events))
+		return transformerBatchResponseT{batchError: err, statusCode: http.StatusInternalServerError}
+	}
+
+	batchResponse := transformerBatchResponseT{responses: make([]transformerResponse, len(events))}
+	for idx, event := range events {
+		pluginOutput, err := plugin.Execute(context.Background(), plugins.NewMessage(event))
+		if err != nil {
+			batchResponse.responses[idx] = bt.markResponseFail(err.Error())
+		}
+		var eventResponse transformerResponse
+		err = mapstructure.Decode(pluginOutput.Data, &eventResponse)
+		if err != nil {
+			batchResponse.responses[idx] = bt.markResponseFail(err.Error())
+		}
+		if eventResponse.Err != "" {
+			batchResponse.responses[idx] = eventResponse
+			continue
+		}
+		if eventResponse.Output == nil && eventResponse.OutputToSource == nil {
+			batchResponse.responses[idx] = bt.markResponseFail(response.SourceTransformerFailedToReadOutput)
+			continue
+		}
+		bt.stats.receivedStat.Count(1)
+		batchResponse.responses[idx] = eventResponse
+	}
+	bt.stats.transformTimerStat.Since(transformStart)
+	bt.stats.receivedStat.Count(len(events))
```

Comment:
Does it make sense to both increment the `receivedStat` in the loop and then at the end with `Count(len(events))`? It looks like a mistake so I thought I'd double check with you.

Reply:
I will remove the inner one